cosmo/monitors/nuv_dark_monitor.py

```diff
@@ -0,0 +1,230 @@
+from .. import SETTINGS
+
+import os
+import yaml
+import cosmo
+import numpy
+import itertools
+from glob import glob
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from matplotlib.ticker import FormatStrFormatter
+from cosmo.monitors.data_models import NUVDarkDataModel
+from cosmo.filesystem import find_files, data_from_exposures, data_from_jitters
+from cosmo.monitor_helpers import absolute_time, explode_df
+from monitorframe.datamodel import BaseDataModel
+from monitorframe.monitor import BaseMonitor
+
+# these imports are so messy i will fix this later
+
+COS_MONITORING = SETTINGS['output']
+
+
+class NUVDarkMonitor(BaseMonitor):
+    data_model = NUVDarkDataModel
+    # TODO: update docs
+    # docs = "https://spacetelescope.github.io/cosmo/monitors.html#<darks>"
+    output = COS_MONITORING
+
+    run = 'monthly'
+
+    def get_data(self):
+        # access data, perform any filtering required for analysis
+        data = self.model.new_data
+        dark_rate_column = []
+        dec_year_column = []
+
+        xlim = [0, 1024]
+        ylim = [0, 1024]
+
+        # parallelize, this is going to get bad when looking at a lot of data
+        for index, row in data.iterrows():
+            subdf = pd.DataFrame({
+                "EXPSTART": row["EXPSTART"], "TIME": [row["TIME"]],
+                "XCORR": [row["XCORR"]], "YCORR": [row["YCORR"]],
+                "TIME_3": [row["TIME_3"]]
+                })
+
+            # this is temporary until i understand james' suggestion to use
+            # df.apply and lambda functions
+            xcorr = subdf["XCORR"][0]
+            ycorr = subdf["YCORR"][0]
+            filtered_xcorr = xcorr[
+                np.where((xcorr > xlim[0]) & (xcorr < xlim[1]))]
+            filtered_ycorr = ycorr[
+                np.where((ycorr > ylim[0]) & (ycorr < ylim[1]))]
+            subdf["XCORR"] = [filtered_xcorr]
+            subdf["YCORR"] = [filtered_ycorr]
+
+            dark_rate_array, dec_year_array = self.calculate_dark_rate(subdf,
+                                                                       xlim,
+                                                                       ylim)
+            dark_rate_column.append(dark_rate_array)
+            dec_year_column.append(dec_year_array)
+
+        data["DARK_RATE"] = dark_rate_column
+        data["DECIMAL_YEAR"] = dec_year_column
+
+        # when the monitor method of the monitor is called, it will
+        # initialize the self.data attribute
+        # with this method and then can be used by the other methods
+        return data
+
+    def calculate_dark_rate(self, dataframe, xlim, ylim):
+        # calculate dark rate for one exposure, with a dataframe with TIME,
+        # XCORR, and YCORR values
+
+        # need to set this somewhere
+        timestep = 25
+        time_bins = dataframe["TIME_3"][0][::timestep]
+
+        counts = np.histogram(dataframe["TIME_3"][0], bins=time_bins)[0]
```

Comment:
```suggestion
        counts = np.histogram(dataframe["TIME"][0], bins=time_bins)[0]
```
I think that if you change the time array you should get what you want. You were essentially asking "how many seconds are in bins of 25 seconds" by accident.